src/main/java/core/basesyntax/db/StorageImpl.java

```diff
@@ -0,0 +1,32 @@
+package core.basesyntax.db;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class StorageImpl implements Storage {
+    private HashMap<String, Integer> fruitMap;
+
+    public StorageImpl() {
+        fruitMap = new HashMap<>();
+    }
+
+    @Override
+    public void add(String fruit, int quantity) {
+        fruitMap.put(fruit, quantity);
+    }
+
+    @Override
+    public void change(String fruit, int newQuantity) {
+        fruitMap.replace(fruit, newQuantity);
+    }
+
+    @Override
+    public int get(String fruit) {
+        return fruitMap.get(fruit);
+    }
+
+    @Override
+    public Map<String, Integer> getAll() {
+        return fruitMap;
```

Comment:
Returning the original `fruitMap` allows external modification of the `StorageImpl` internal state. To encapsulate the internal state, consider returning a copy of the map, for example using `new HashMap<>(fruitMap)`.